wdl-format/src/token/post.rs

```diff
@@ -258,21 +375,131 @@ impl Postprocessor {
         }
     }
 
+    /// Flushes the `in_stream` buffer to the `out_stream`.
+    fn flush(
+        &mut self,
+        in_stream: &TokenStream<PreToken>,
+        out_stream: &mut TokenStream<PostToken>,
+        config: &Config,
+    ) {
+        assert!(!self.interrupted);
+        assert!(self.position == LinePosition::StartOfLine);
+        let mut post_buffer = TokenStream::<PostToken>::default();
+        let mut pre_buffer = in_stream.iter().peekable();
+        let starting_indent = self.indent_level;
+        while let Some(token) = pre_buffer.next() {
+            let next = pre_buffer.peek().copied();
+            self.step(token.clone(), next, &mut post_buffer);
+        }
+
+        // If all lines are short enough, we can just add the post_buffer to the
+        // out_stream and be done.
+        if config.max_line_length().is_none()
+            || post_buffer.max_width(config) <= config.max_line_length().unwrap()
+        {
+            out_stream.extend(post_buffer);
+            return;
+        }
+
+        // At least one line in the post_buffer is too long.
+        // We iterate through the in_stream to find potential line breaks,
+        // and then we iterate through the in_stream again to actually insert
+        // them in the proper places.
+
+        let max_length = config.max_line_length().unwrap();
+
+        let mut potential_line_breaks: Vec<usize> = Vec::new();
```

Comment:
Could this just start off as a `HashSet` rather than `Vec` that gets remade into a `HashSet`?